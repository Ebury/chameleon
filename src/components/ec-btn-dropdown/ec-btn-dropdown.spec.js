import { mount } from '@vue/test-utils';

import EcBtnDropdown from './ec-btn-dropdown.vue';

describe('EcBtnDropdown', () => {
  function mountBtnDropdown(props, mountOpts) {
    return mount(EcBtnDropdown, {
      props: {
        buttonText: 'Convert & Pay',
        ...props,
      },
      ...mountOpts,
    });
  }

  let items = [
    { value: 'Spot', text: 'Spot' },
    { value: 'Convert', text: 'Convert' },
  ];

  it('should render as expected', () => {
    const wrapper = mountBtnDropdown();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with a list of options on the dropdown search', () => {
    const wrapper = mountBtnDropdown({ items });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with both buttons disabled', () => {
    const wrapper = mountBtnDropdown({
      isDisabled: true,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use given list-data-test prop', () => {
    const wrapper = mountBtnDropdown({
      items,
      listDataTest: 'my-test-list',
    });

    expect(wrapper.findByDataTest('my-test-list').element).toMatchSnapshot();
  });

  it('should emit the click event when the user clicks on the primary button', async () => {
    const wrapper = mountBtnDropdown();

    await wrapper.findByDataTest('ec-btn-dropdown__btn').trigger('click');
    expect(wrapper.emitted('click').length).toBe(1);
  });

  it('should emit the change event when the user clicks on an item of the dropdown list', async () => {
    const wrapper = mountBtnDropdown({ items });

    await wrapper.findByDataTest('ec-btn-dropdown__dropdown-btn').trigger('click');
    await wrapper.findAllByDataTest('ec-dropdown-search__item').at(0).trigger('click');

    expect(wrapper.emitted('change').length).toBe(1);
    expect(wrapper.emitted('change')[0]).toEqual([items[0]]);
  });

  it('should render with a ctabtnlink slot given', () => {
    const wrapper = mountBtnDropdown({
      slots: {
        ctabtnlink: `
          <template #ctabtnlink="{buttonText}">
          <router-link
            class="ec-btn__text-link"
            :to="items[0].to"
          >{{ buttonText }}
          </router-link>
        </template>`,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render with an item slot given', () => {
    items = [
      {
        href: '/trade/drawdown/',
        value: 'drawdown',
        text: 'Drawdown',
        disabled: false,
      },
    ];

    const wrapper = mountBtnDropdown(
      { items },
      {
        slots: {
          item: `
        <template #item="{ item }">
        <a
          class="ec-btn-dropdown__item-link"
          href="convert-and-pay"
        >Convert & Pay</a>
      </template>`,
        },
      },
    );
    expect(wrapper.element).toMatchSnapshot();
  });
});
