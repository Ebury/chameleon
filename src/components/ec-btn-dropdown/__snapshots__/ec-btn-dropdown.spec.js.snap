// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcBtnDropdown should emit a click event when the CTA button is clicked 1`] = `
<button
  class="ec-btn-dropdown__btn ec-btn ec-btn--md ec-btn--rounded ec-btn--primary ec-btn--primary-reverse"
  data-test="ec-btn-dropdown__btn ec-btn"
  type="button"
>
  <!--v-if-->
  <span
    class="ec-btn__text"
    data-test="ec-btn__text"
  >
    
    Convert & Pay
    
  </span>
  <!--v-if-->
</button>
`;

exports[`EcBtnDropdown should render as expected 1`] = `
<div
  class="ec-btn-dropdown"
  data-test="ec-btn-dropdown"
>
  <button
    class="ec-btn-dropdown__btn ec-btn ec-btn--md ec-btn--rounded ec-btn--primary ec-btn--primary-reverse"
    data-test="ec-btn-dropdown__btn ec-btn"
    type="button"
  >
    <!--v-if-->
    <span
      class="ec-btn__text"
      data-test="ec-btn__text"
    >
      
      Convert & Pay
      
    </span>
    <!--v-if-->
  </button>
  <div
    class="ec-dropdown-search"
    data-test="ec-dropdown-search"
  >
    <ec-stub
      autosize="min"
      class="ec-dropdown-search__trigger"
      data-test="ec-popover-dropdown-search ec-stub ec-popover-stub"
      disabled="false"
      distance="8"
      placement="bottom-end"
      popperclass="ec-dropdown-search__popover"
      shift="false"
      shown="false"
    >
      
      
      <button
        class="ec-btn-dropdown__dropdown-btn ec-btn ec-btn--md ec-btn--rounded ec-btn--icon-only ec-btn--primary ec-btn--primary-reverse"
        data-test="ec-btn-dropdown__dropdown-btn ec-btn"
        type="submit"
      >
        <svg
          class="ec-icon ec-btn__icon"
          data-test="ec-btn__icon"
          height="22"
          width="22"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
        <!--v-if-->
        <!--v-if-->
      </button>
      
      
      
      <div>
        <ul
          class="ec-dropdown-search__item-list"
          data-test="ec-dropdown-search__item-list"
        >
          <!--v-if-->
          <!--v-if-->
          
          
          
          
        </ul>
      </div>
      
    </ec-stub>
  </div>
</div>
`;

exports[`EcBtnDropdown should render with a 'to' prop 1`] = `
<button
  class="ec-btn-dropdown__btn ec-btn ec-btn--md ec-btn--rounded ec-btn--primary ec-btn--primary-reverse"
  data-test="ec-btn-dropdown__btn ec-btn"
  type="button"
>
  <!--v-if-->
  <span
    class="ec-btn__text"
    data-test="ec-btn__text"
  >
    
    Convert & Pay
    
  </span>
  <!--v-if-->
</button>
`;

exports[`EcBtnDropdown should render with a list of options on the dropdown search 1`] = `
<div
  class="ec-btn-dropdown"
  data-test="ec-btn-dropdown"
>
  <button
    class="ec-btn-dropdown__btn ec-btn ec-btn--md ec-btn--rounded ec-btn--primary ec-btn--primary-reverse"
    data-test="ec-btn-dropdown__btn ec-btn"
    type="button"
  >
    <!--v-if-->
    <span
      class="ec-btn__text"
      data-test="ec-btn__text"
    >
      
      Convert & Pay
      
    </span>
    <!--v-if-->
  </button>
  <div
    class="ec-dropdown-search"
    data-test="ec-dropdown-search"
  >
    <ec-stub
      autosize="min"
      class="ec-dropdown-search__trigger"
      data-test="ec-popover-dropdown-search ec-stub ec-popover-stub"
      disabled="false"
      distance="8"
      placement="bottom-end"
      popperclass="ec-dropdown-search__popover"
      shift="false"
      shown="false"
    >
      
      
      <button
        class="ec-btn-dropdown__dropdown-btn ec-btn ec-btn--md ec-btn--rounded ec-btn--icon-only ec-btn--primary ec-btn--primary-reverse"
        data-test="ec-btn-dropdown__dropdown-btn ec-btn"
        type="submit"
      >
        <svg
          class="ec-icon ec-btn__icon"
          data-test="ec-btn__icon"
          height="22"
          width="22"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
        <!--v-if-->
        <!--v-if-->
      </button>
      
      
      
      <div>
        <ul
          class="ec-dropdown-search__item-list"
          data-test="ec-dropdown-search__item-list"
        >
          <!--v-if-->
          <!--v-if-->
          
          
          <li
            class="ec-dropdown-search__item"
            data-test="ec-dropdown-search__item ec-dropdown-search__item--0 ec-mock ec-tooltip-mock"
            title="Spot"
          >
            
            <ec-stub
              class="ec-btn-dropdown__item-link"
              data-test="ec-stub router-link-stub"
            >
              
              Spot
              
            </ec-stub>
            
          </li>
          <li
            class="ec-dropdown-search__item"
            data-test="ec-dropdown-search__item ec-dropdown-search__item--1 ec-mock ec-tooltip-mock"
            title="Convert"
          >
            
            <ec-stub
              class="ec-btn-dropdown__item-link"
              data-test="ec-stub router-link-stub"
            >
              
              Convert
              
            </ec-stub>
            
          </li>
          
          
        </ul>
      </div>
      
    </ec-stub>
  </div>
</div>
`;

exports[`EcBtnDropdown should render with both buttons disabled 1`] = `
<div
  class="ec-btn-dropdown"
  data-test="ec-btn-dropdown"
>
  <button
    class="ec-btn-dropdown__btn ec-btn ec-btn--md ec-btn--rounded ec-btn--primary ec-btn--primary-reverse"
    data-test="ec-btn-dropdown__btn ec-btn"
    disabled=""
    type="button"
  >
    <!--v-if-->
    <span
      class="ec-btn__text"
      data-test="ec-btn__text"
    >
      
      Convert & Pay
      
    </span>
    <!--v-if-->
  </button>
  <div
    class="ec-dropdown-search"
    data-test="ec-dropdown-search"
  >
    <ec-stub
      autosize="min"
      class="ec-dropdown-search__trigger"
      data-test="ec-popover-dropdown-search ec-stub ec-popover-stub"
      disabled="true"
      distance="8"
      placement="bottom-end"
      popperclass="ec-dropdown-search__popover"
      shift="false"
      shown="false"
    >
      
      
      <button
        class="ec-btn-dropdown__dropdown-btn ec-btn ec-btn--md ec-btn--rounded ec-btn--icon-only ec-btn--primary ec-btn--primary-reverse"
        data-test="ec-btn-dropdown__dropdown-btn ec-btn"
        disabled=""
        type="submit"
      >
        <svg
          class="ec-icon ec-btn__icon"
          data-test="ec-btn__icon"
          height="22"
          width="22"
        >
          <use
            xlink:href="#ec-simple-arrow-drop-down"
          />
        </svg>
        <!--v-if-->
        <!--v-if-->
      </button>
      
      
      
      <div>
        <ul
          class="ec-dropdown-search__item-list"
          data-test="ec-dropdown-search__item-list"
        >
          <!--v-if-->
          <!--v-if-->
          
          
          
          
        </ul>
      </div>
      
    </ec-stub>
  </div>
</div>
`;

exports[`EcBtnDropdown should use given list-data-test prop 1`] = `
<ul
  class="ec-dropdown-search__item-list"
  data-test="ec-dropdown-search__item-list my-test-list"
>
  <!--v-if-->
  <!--v-if-->
  
  
  <li
    class="ec-dropdown-search__item"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--0 ec-mock ec-tooltip-mock"
    title="Spot"
  >
    
    <ec-stub
      class="ec-btn-dropdown__item-link"
      data-test="ec-stub router-link-stub"
    >
      
      Spot
      
    </ec-stub>
    
  </li>
  <li
    class="ec-dropdown-search__item"
    data-test="ec-dropdown-search__item ec-dropdown-search__item--1 ec-mock ec-tooltip-mock"
    title="Convert"
  >
    
    <ec-stub
      class="ec-btn-dropdown__item-link"
      data-test="ec-stub router-link-stub"
    >
      
      Convert
      
    </ec-stub>
    
  </li>
  
  
</ul>
`;

exports[`EcBtnDropdown when the CTA menu item is disabled should be a non clickable DOM element 1`] = `
<li
  class="ec-dropdown-search__item ec-dropdown-search__item--is-disabled"
  data-test="ec-dropdown-search__item ec-dropdown-search__item--0 ec-mock ec-tooltip-mock"
  title="Convert & Pay"
>
  
  <span>
    Convert & Pay
  </span>
  
</li>
`;
